java/client/src/main/java/glide/connectors/handlers/ReadHandler.java

```diff
@@ -32,8 +32,20 @@ public void channelRead(@NonNull ChannelHandlerContext ctx, @NonNull Object msg)
     /** Handles uncaught exceptions from {@link #channelRead(ChannelHandlerContext, Object)}. */
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
-        Logger.log(ERROR, "read handler", () -> "=== exceptionCaught " + ctx + " " + cause);
+        if (cause instanceof MessageHandler.MessageCallbackException) {
+            Logger.log(
+                    ERROR,
+                    "read handler",
+                    () -> "=== Exception thrown from pubsub callback " + ctx,
+                    cause.getCause());
 
+            // Mimic the behavior of if this got thrown by a user thread. Print to stderr,
+            cause.printStackTrace();
+
+            // Unwrap. Only works for Exceptions and not Errors.
+            throw ((MessageHandler.MessageCallbackException) cause).getCause();
```

Comment:
You need IT (and ideally UT) for this to prove this works fine

Reply:
There's a test for this in the integration tests:
pubsub_test_callback_exception.

It's disabled though because it causes Netty to throw from a background thread.